include/ibamr/vc_ins_utilities.h

```diff
@@ -0,0 +1,411 @@
+                                        SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                        const int cycle_num,
+                                        const double time,
+                                        const double current_time,
+                                        const double new_time);
+
+    /*!
+     * Set the viscosity based on the current level set information for two-phase flows.
+     */
+    void setViscosityPatchData2PhaseFlows(int mu_idx,
+                                          SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                          SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                          const int cycle_num,
+                                          const double time,
+                                          const double current_time,
+                                          const double new_time);
+
+    /*!
+     * Set the viscosity based on the current level set information for three-phase flows.
+     */
+    void setViscosityPatchData3PhaseFlows(int mu_idx,
+                                          SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                          SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                          const int cycle_num,
+                                          const double time,
+                                          const double current_time,
+                                          const double new_time);
+
+    /*!
+     * Name of this object.
+     */
+    std::string d_object_name;
+
+    /*!
+     * Pointer to advection-diffusion solver.
+     */
+    SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> d_adv_diff_solver;
+
+    /*!
+     * Level set variable.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_ls_gas_var, d_ls_solid_var;
+
+    /*!
+     * Density.
+     */
+    double d_rho_liquid, d_rho_gas, d_rho_solid;
+
+    /*!
+     * Viscosity.
+     */
+    double d_mu_liquid, d_mu_gas, d_mu_solid;
+
+    /*!
+     * Number of interface cells over which to smooth the material properties.
+     */
+    double d_num_gas_interface_cells, d_num_solid_interface_cells;
+
+    /*!
+     * Whether or not to set viscosity in the solid region.
+     */
+    bool d_set_mu_solid;
+
+    /*!
+     * Number of phases. Valid options are 2 and 3.
+     */
+    int d_num_phases = 2;
+
+}; // SetFluidProperties
+
+/*!
+ * \brief The GravityForcing class provides an implementation of gravity force.
+ * This class can be utilized to set the gravity force \f$ \rho g \f$ throughout the entire domain with
+ * the option grav_type = "FULL" or specifically within the liquid and gas regions (excluding the solid phase)
+ * with the option grav_type = "FLOW".
+ */
```

Comment:
`grav_type` is no longer a parameter to the system.
```suggestion
/*!
 * \brief The GravityForcing class provides an implementation of gravity force.
 * This class can be utilized to set the gravity force \f$ \rho g \f$ throughout the entire domain or specifically within the liquid and gas regions (excluding the solid phase)
 */
```